### DISCUSSION TITLE
Settings for different cameras to output to different directories

### DISCUSSION THREAD

**Discussion by bisechen** [Mainman002/TMG-Camera-Tools#5] [Ideas]
The output of this ADDON seems to be unable to specify each camera in a different output directory?

I see that there is Output in the UI design, but in fact, it is not possible to make settings for each camera, for example, the Render results of camera A are stored in the A directory, and the B camera is stored in the B directory.

Hopefully there will be enhancements like this, which would be very useful.

In addition, a function of batch calculation can be added, which can be automatically calculated and stored separately with just one click, which can greatly reduce the time waiting on the computer. Although we can make some batch files by ourselves, we can set them directly in the UI. would be the most convenient.

**Comment by Mainman002** [comment]
Oo that's a good request! Yes currently it only exports to a single directory. . . Hmm I could probably make that work ;)

**Comment by Mainman002** [comment]
I managed to add the render path to a camera custom property so it can be changed independently based on the active camera :) 
In it's current state the active camera can't be triggered to being active by a keyframe so the output path won't change automatically during an animation (unless drivers can trigger a camera change from my add-on?) If anything at least it's another epic step forward ;)

![image](https://user-images.githubusercontent.com/11281480/180371238-43d8bfad-5485-4a1a-be29-a4dddead79b6.png)

**Reply by bisechen** [reply]
oh! You are awesome, I just downloaded your new change that stores the camera's Render into their own directory!

**Reply by bisechen** [reply]
I found this default // the same path as the file is great, but the saved result doesn't seem to have a file name?

**Reply by Mainman002** [reply]
That could be tricky. Animations tend to name themselves based on frame number, but single images should be able to use a name value? I'll have to play around with a few things and see if I can get that part working ;)

[EDIT]
Oo cool yes I can add a file name option that should work :)

**Reply by bisechen** [reply]
I've been using your great work for a few hours now, the only downside now is the need to press one camera at a time to Render, but it's a lot easier than having to change paths... XD

And I found that even if I append these cameras into another file, these path settings still exist~

**Reply by Mainman002** [reply]
Thank you for your kind words, it brings me great joy to see someone else using it too :)

**Comment by Mainman002** [comment]
The tricky thing is most of my panel settings are default blender buttons just placed in what I hope is an easier location / layout to find more efficiently. I store custom properties that I like / modified on the camera object data directly. In theory saving cameras to the asset library might be possible, but I'd have to play around and see if that's an acceptable work around for presets ( I'd love to have custom presets, but I haven't created that type of setup yet ) 

![image](https://user-images.githubusercontent.com/11281480/180373759-3dd7c6dd-47d5-4a17-8f96-0a2a30f44dfe.png)

**Comment by Mainman002** [comment]
So I managed to get the compositor to export an image to the directory, the trick is that when you render an animation it will render twice the frames. . . Tricky little bugger to get working properly but it's getting there :)

You'll have to add a File Output node to the compositor, hook into the Render Layers Image output, and enable Use_Nodes, but the rest is automated.

![image](https://user-images.githubusercontent.com/11281480/180437913-44023400-a4b9-4776-bebd-58800e0d5ed0.png)
